src/HotChocolate/Core/src/Types/Configuration/TypeInitializer.cs

```diff
@@ -541,6 +556,29 @@ bool CompleteType(RegisteredType registeredType)
             return _errors.Count == 0;
         }
 
+        private IEnumerable<RegisteredType> GetSelfReferencingTypes(
+            IEnumerable<RegisteredType> registeredTypes,
+            TypeDependencyKind kind)
+        {
+            foreach (var type in registeredTypes)
+            {
+                IEnumerable<IExtendedType> dependencies = type.Dependencies
+                    .Where(x => x.Kind == kind)
+                    .Select(x => x.TypeReference)
+                    .OfType<ExtendedTypeReference>()
+                    .Select(x => x.Type);
+
+                IEnumerable<IExtendedType> references = type.References
+                    .OfType<ExtendedTypeReference>()
+                    .Select(x => x.Type);
+
+                if (dependencies.Intersect(references).Any())
+                {
```

Comment:
This could also initialize other dependencies that should not yet be resovler
we may want to do a deep check here to see if the missing dependencies really form a cycle.

Reply:
I think we should also have a look at why the dependencies are hard dependencies.

Meaning in the offset case we have 

ConnectionA->EdgeA->TypeA->ConnectionB->EdgeB->TypeB->ConnectionA

each of the dependency demands completion. We also have a concept of weak dependencies. So what actually should happen is that field dependencies are weak dependencies... meaning the pointer from TypeA to ConnectionB should be weak.